src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extension.cs

```diff
@@ -63,6 +64,321 @@ internal override string ExtensionName
+                }
+                else if (typeParameter.HasValueTypeConstraint)
+                {
+                    typeConstraintStrings.Add("class System.ValueType");
+                }
+
+                typeConstraintStrings.Sort(StringComparer.Ordinal); // Actual order doesn't matter - just want to be deterministic
+
+                builder.Append('(');
+                for (int i = 0; i < typeConstraintStrings.Count(); i++)
+                {
+                    if (i > 0)
+                    {
+                        builder.Append(", ");
+                    }
+
+                    builder.Append(typeConstraintStrings[i]);
+                }
+
+                typeConstraintStrings.Free();
+                builder.Append(") ");
+            }
+
+            static void appendArrayType(ArrayTypeSymbol array, StringBuilder builder)
+            {
+                appendType(array.ElementType, builder);
+                builder.Append('[');
+                for (int i = 0; i < array.Rank; i++)
+                {
+                    if (i > 0)
+                    {
+                        builder.Append(',');
+                    }
+                }
+
+                builder.Append(']');
+            }
+
+            static void appendFunctionPointerType(FunctionPointerTypeSymbol functionPointer, StringBuilder builder)
+            {
+                builder.Append("method ");
+
+                // When calling convention is a single one of the four special calling conventions, we just use flags
+                // Otherwise, we use "unmanaged" flag and also add return modifiers below
+                var callingConvention = functionPointer.Signature.CallingConvention switch
+                {
+                    Cci.CallingConvention.Default => null, // managed is the default
+                    Cci.CallingConvention.Unmanaged => "unmanaged ",
+                    Cci.CallingConvention.CDecl => "unmanaged cdecl ",
+                    Cci.CallingConvention.Standard => "unmanaged stdcall ",
+                    Cci.CallingConvention.ThisCall => "unmanaged thiscall ",
+                    Cci.CallingConvention.FastCall => "unmanaged fastcall ",
+                    _ => throw ExceptionUtilities.UnexpectedValue(functionPointer.Signature.CallingConvention)
+                };
+
+                builder.Append(callingConvention);
+
+                appendType(functionPointer.Signature.ReturnType, builder);
+                if (functionPointer.Signature.RefKind != RefKind.None)
+                {
+                    builder.Append('&');
+                    appendModifiers(functionPointer.Signature.RefCustomModifiers, builder);
+                }
+                else
+                {
+                    appendModifiers(functionPointer.Signature.ReturnTypeWithAnnotations.CustomModifiers, builder);
```

Comment:
>appendModifiers(functionPointer.Signature.ReturnTypeWithAnnotations.CustomModifiers, builder); [](http://example.com/codeflow?start=20&length=94)

This doesn't look right. There could be ```ReturnTypeWithAnnotations.CustomModifiers``` even when ```functionPointer.Signature.RefKind != RefKind.None```

Reply:
I don't think that's possible. See logic in `FunctionPointerMethodSymbol.CreateFromSource`